ethdb/privateapi/ethbackend.go

```diff
@@ -121,3 +123,11 @@ func (s *EthBackendServer) ProtocolVersion(_ context.Context, _ *remote.Protocol
 func (s *EthBackendServer) ClientVersion(_ context.Context, _ *remote.ClientVersionRequest) (*remote.ClientVersionReply, error) {
 	return &remote.ClientVersionReply{NodeName: common.MakeName("erigon", params.Version)}, nil
 }
+
+func (s *EthBackendServer) NodeInfo(_ context.Context, r *remote.NodesInfoRequest) (*remote.NodesInfoReply, error) {
+	nodesInfo, err := s.eth.NodesInfo(int(r.Limit))
+	if err != nil {
+		return nil, err
+	}
+	return nodesInfo, nil
```

Comment:
I believe this can be simplified to 
```suggestion
	nodesInfo, err := s.eth.NodesInfo(int(r.Limit))
	return nodesInfo, err
```
as s.eth.nodesInfo on err returns nodesInfo == nil
The if branch is pleonastic

Reply:
Yes, it can be simplified further into `return s.eth.NodesInfo(int(r.Limit))`